lib/sql-string.js

```diff
@@ -35,6 +35,11 @@ function escape(val, timeZone, dialect, format) {
     break;
   }
 
+  // moment
+  if (val.toDate) {
```

Comment:
I highly argue against that. Interfaces in JavaScript and TypeScript are based around duck-typing, because there is no `implements` like in Java or overloading. So the method name is the contract. If you call your method `toDate`, it should return a `Date`. Event the standard JavaScript objects use loose interface like this, for example `Promise.resolve()`. If your object has a `.then()` method, it is expected to accept a promise handler and return a new promise, without any `implements` declaration. The only contract is the method name here.